core-kotlin/src/main/java/com/amplifyframework/kotlin/datastore/DataStore.kt

```diff
@@ -73,33 +73,45 @@ interface DataStore {
 
     /**
      * Observe all changes to items in the DataStore.
+     * This function suspends until observation has been setup.
+     * Once the observation begins, the function resumes and returns a Flow.
+     * The returned Flow represents a stream of changes to the DataStore.
      * @return A flow of changes to the items in the DataStore
      */
     @ExperimentalCoroutinesApi
     @Throws(DataStoreException::class)
-    fun observe(): Flow<DataStoreItemChange<out Model>>
+    suspend fun observe(): Flow<DataStoreItemChange<out Model>>
 
     /**
      * Observe the DataStore for changes to a particular item.
+     * This function suspends until observation has been setup.
+     * Once the observation begins, the function resumes and returns a Flow.
+     * The returned Flow represents a stream of changes to the DataStore.
      * @param itemClass The class of the model being observed
      * @param uniqueId The ID of the item being observed
      * @return A flow of changes to the requested model
      */
     @ExperimentalCoroutinesApi
     @Throws(DataStoreException::class)
-    fun <T : Model> observe(itemClass: Class<T>, uniqueId: String): Flow<DataStoreItemChange<T>>
+    suspend fun <T : Model> observe(
+        itemClass: Class<T>,
+        uniqueId: String
```

Comment:
Maybe call this `itemId`?  Just a thought, though I don't feel strongly one way or the other.

```suggestion
        itemId: String
```

Reply:
Will do! Update coming ...